### PR TITLE
 Introduce mutliclass support for SVC models (one-vs-one decision fucntion output)

### PR DESCRIPTION
**Pull request by izeigerman**

### CODE REVIEW

File: tests/assemblers/test_svm.py [hunk 1]
```diff
@@ -125,7 +107,67 @@ def test_unknown_kernel():
...
+                    ast.BinNumOpType.MUL),
+                ast.BinNumOpType.ADD),
+            ast.BinNumExpr(
+                kernels[1],
+                ast.NumVal(1.0),
+                ast.BinNumOpType.MUL),
+            ast.BinNumOpType.ADD)])
+
+    # expected = _create_expected_single_output_ast(
+    #     estimator, [kernel_ast(1.0), kernel_ast(2.0), kernel_ast(3.0)])
```
[the remark this answers is not in the document]
Comment (izeigerman) [reply tests/assemblers/test_svm.py]: oops, forgot to do that. Thanks!

File: m2cgen/assemblers/svm.py [hunk 2]
```diff
@@ -51,15 +57,53 @@ def _assemble_single_output(self):
...
+        intercept = self.model.intercept_
+
+        n_support = self.model.n_support_
+        n_support_len = len(n_support)
+
+        kernel_exprs = self._apply_kernel(support_vectors, to_reuse=True)
+
+        support_ranges = []
+        for i in range(n_support_len):
+            range_start = sum(n_support[0:i])
```
[the remark this answers is not in the document]
Comment (izeigerman) [reply m2cgen/assemblers/svm.py]: oh, right. Thanks!

File: tests/assemblers/test_svm.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (izeigerman) [reply tests/assemblers/test_svm.py]: Fixed

File: m2cgen/interpreters/interpreter.py [hunk 3]
```diff
@@ -33,10 +33,14 @@ def _do_interpret(self, expr, **kwargs):
...
             if isinstance(expr, ast.TransparentExpr):
-                return self._do_interpret(expr.expr, **kwargs)
+                reuse = True if expr.to_reuse else None
+                return self._do_interpret(expr.expr, to_reuse=reuse, **kwargs)
             raise
 
-        if not expr.to_reuse:
+        # Note that the reuse flag passed in the arguments has a higher
+        # precedence than one specified in the expression.
+        expr_to_reuse = to_reuse if to_reuse is not None else expr.to_reuse
```
[the remark this answers is not in the document]
Comment (izeigerman) [reply m2cgen/interpreters/interpreter.py]: Fixed

File: m2cgen/assemblers/svm.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (izeigerman) [reply m2cgen/assemblers/svm.py]: Fixed

**Comment by krinart** [comment]
Impressive job!

**Pull request merged**